### PR TITLE
APIM 3921: Add Api service to portal-next

### PR DESCRIPTION
**Pull request by jourdiw**
## Issue

https://gravitee.atlassian.net/browse/APIM-3921

## Description

- create config service as a store
- initialize api service + tests
- integrate service into the catalog component

With public + published APIs returned:

Screenshot 2024-04-11 at 17 35 47

With no apis that are public + published:

Screenshot 2024-04-10 at 09 45 11

## Additional context

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-fsrwjzjcxc.chromatic.com)

### CODE REVIEW

File: gravitee-apim-portal-webui-next/src/services/config.service.ts [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+ * limitations under the License.
+ */
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { map, Observable, switchMap } from 'rxjs';
+
+interface Config {
+  baseURL: string;
+  environmentId: string;
+  organizationId: string;
```
Comment (phiz71) [inline gravitee-apim-portal-webui-next/src/services/config.service.ts]: to be removed ?

File: gravitee-apim-portal-webui-next/src/services/config.service.ts [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    const baseURL = config.baseURL;
+    const orgIndex = baseURL.indexOf('/environments/');
+    if (orgIndex >= 0) {
+      const subPathWithOrg = baseURL.substring(orgIndex, baseURL.length);
+      const splitArr = subPathWithOrg.split('/');
+      if (splitArr.length >= 3) {
+        return splitArr[2];
+      }
+    }
+    return '';
```
Comment (phiz71) [inline gravitee-apim-portal-webui-next/src/services/config.service.ts]: Maybe replace any reference to `org` by `envs` or `environments`

**Review by phiz71 — CHANGES_REQUESTED** [review]
LGTM, some suggestion

File: gravitee-apim-portal-webui-next/src/services/config.service.ts (hunk 1 above)
Comment (jourdiw) [reply gravitee-apim-portal-webui-next/src/services/config.service.ts]: It's what is currently being returned by `/ui/bootstrap`. In the other portal and in portal-next, our local `config.json` can specify the `environmentId`. 

Perhaps I can add a comment to explain the interface?

**Review by phiz71 — APPROVED** [review]

**Review by phiz71 — APPROVED** [review]

**Pull request merged**